diff --git a/bsp/board/bl616dk/board.c b/bsp/board/bl616dk/board.c
index e7a8ce56..428a0e2b 100644
--- a/bsp/board/bl616dk/board.c
+++ b/bsp/board/bl616dk/board.c
@@ -212,7 +212,7 @@ void board_init(void)
 
     flag = bflb_irq_save();
 
-    ret = bflb_flash_init();
+    //ret = bflb_flash_init();
 
     system_clock_init();
     peripheral_clock_init();
diff --git a/bsp/board/bl616dk/fw_header.c b/bsp/board/bl616dk/fw_header.c
index 742cf820..76d1c141 100644
--- a/bsp/board/bl616dk/fw_header.c
+++ b/bsp/board/bl616dk/fw_header.c
@@ -6,7 +6,7 @@ __attribute__((section(".fw_header"))) struct bootheader_t fw_header = {
     /*flash config */
     .flash_cfg.magiccode = 0x47464346,
     .flash_cfg.cfg.ioMode = 0x11,               /*!< Serail flash interface mode,bit0-3:IF mode,bit4:unwrap */
-    .flash_cfg.cfg.cReadSupport = 0x00,         /*!< Support continuous read mode,bit0:continuous read mode support,bit1:read mode cfg */
+    .flash_cfg.cfg.cReadSupport = 0x01,         /*!< Support continuous read mode,bit0:continuous read mode support,bit1:read mode cfg */
     .flash_cfg.cfg.clkDelay = 0x01,             /*!< SPI clock delay,bit0-3:delay,bit4-6:pad delay */
     .flash_cfg.cfg.clkInvert = 0x01,            /*!< SPI clock phase invert,bit0:clck invert,bit1:rx invert,bit2-4:pad delay,bit5-7:pad delay */
     .flash_cfg.cfg.resetEnCmd = 0x66,           /*!< Flash enable reset command */
